src/providers/ses.rs

```diff
@@ -0,0 +1,89 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::boxed::Box;
+
+use rusoto_core::{reactor::RequestDispatcher, Region};
+use rusoto_credential::StaticProvider;
+use rusoto_ses::{Content, Destination, Message, SendEmailRequest, Ses, SesClient};
+
+use super::{Provider, ProviderError};
+use settings::Settings;
+
+pub struct SesProvider
+{
+  client: Box<Ses>,
+  sender: String,
+}
+
+impl SesProvider
+{
+  pub fn new(settings: &Settings) -> SesProvider
+  {
+    let region = settings
+      .ses
+      .region
+      .parse::<Region>()
+      .expect("invalid region");
+
+    let client: Box<Ses> = if let Some(ref keys) = settings.ses.keys {
+      let creds = StaticProvider::new(keys.access.to_string(), keys.secret.to_string(), None, None);
+      Box::new(SesClient::new(RequestDispatcher::default(), creds, region))
+    } else {
+      Box::new(SesClient::simple(region))
+    };
+
+    SesProvider {
+      client,
+      sender: settings.sender.clone(),
+    }
+  }
+}
+
+impl Provider for SesProvider
+{
+  fn send(
+    &self,
+    to: String,
+    cc: Vec<String>,
+    subject: String,
+    body_text: String,
+    body_html: Option<String>,
+  ) -> Result<String, ProviderError>
+  {
+    let mut destination = Destination::default();
+    destination.to_addresses = Some(vec![to]);
+    if cc.len() > 0 {
+      destination.cc_addresses = Some(cc);
+    }
+
+    let mut message = Message::default();
+    message.subject.data = subject;
+    message.body.text = Some(Content {
+      charset: None,
```

Comment:
Should we use an explicit charset here, or does it default to doing something sensible?

Reply:
The relevant docs are these:

https://rusoto.github.io/rusoto/rusoto_ses/struct.Content.html

I have an issue open for unicode support in #9 and there's no immediate need to send unicode-enabled email right now, so I figured it was okay to leave the default behaviour in this PR and come back to implement uncode properly later on, in an everything-definitely-works-end-to-end fashion.

Reply:
(but I'll add a comment to the issue to remind me that this bit needs doing too)